gcp-firebase/src/main/java/com/baeldung/gcp/firebase/auth/FirebaseAuthConfiguration.java

```diff
@@ -0,0 +1,34 @@
+package com.baeldung.gcp.firebase.auth;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.io.Resource;
+
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.auth.FirebaseAuth;
+
+@Configuration
+public class FirebaseAuthConfiguration {
+
+    @Value("classpath:/private-key.json")
+    private Resource privateKey;
+
+    @Bean
+    public FirebaseAuth firebaseAuth() throws IOException {
+        InputStream credentials = new ByteArrayInputStream(privateKey.getContentAsByteArray());
+        FirebaseOptions firebaseOptions = FirebaseOptions.builder()
+            .setCredentials(GoogleCredentials.fromStream(credentials))
+            .build();
+
+        FirebaseApp firebaseApp = FirebaseApp.initializeApp(firebaseOptions);
```

Comment:
I'd expect FirebaseApp to be a separate bean too, since in a real app it might need to be reused to create other objects too beyond FirebaseAuth

Reply:
Review incorporated.
Made the changes in the article as well and explained the reasoning to have separate beans.